Dynatrace/dynatrace-operator
# Add k8s.cluster.name to enrichment + renames

## Description
[K8S-9858](https://dt-rnd.atlassian.net/browse/K8S-9858)

Adds the `k8s.workload.*` renames:
- [Update init container to add new props alongside old ones](https://github.com/Dynatrace/dynatrace-operator/commit/974d02539901cf1bb000d2198f3cbf92b11f32da) 

Adds the `k8s.cluster.name` to the metadata enrichment files:
- [Updated the webhook to add a new ENV based on the dk.Status](https://github.com/Dynatrace/dynatrace-operator/commit/470a273356e9d4c23bfa3d1bf4974b293fa48c51)
  - Wanted to add it to the init-secret (why [this ](https://github.com/Dynatrace/dynatrace-operator/commit/8fbce49bd4cc1a87b62a0052a150fa9da90e29f7)was done), but that is mainly for the OA injection, so decided against it. 	
- [Updated the init container to use the new ENV ](https://github.com/Dynatrace/dynatrace-operator/commit/680f88386b397a62b8a562175b31c51827001325)

## How can this be tested?

1. Deploy DK with metadataEnrichment enabled
2. Deploy sample that will get injected
3. Look for the new envvars on the Pod
4. Check Pod's fs to see if the properties files are correct
   - `/var/lib/dynatrace/enrichment/dt_metadata.properties`
   - `/var/lib/dynatrace/enrichment/dt_metadata.json`
